maat-orchestration/src/main/java/uk/gov/justice/laa/crime/orchestration/mapper/IncomeEvidenceMapper.java

```diff
@@ -0,0 +1,241 @@
+import uk.gov.justice.laa.crime.common.model.evidence.*;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.FinancialAssessmentIncomeEvidence;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.MaatApiAssessmentResponse;
+import uk.gov.justice.laa.crime.common.model.meansassessment.maatapi.MaatApiUpdateAssessment;
+import uk.gov.justice.laa.crime.enums.EmploymentStatus;
+import uk.gov.justice.laa.crime.enums.MagCourtOutcome;
+import uk.gov.justice.laa.crime.exception.ValidationException;
+import uk.gov.justice.laa.crime.orchestration.dto.WorkflowRequest;
+import uk.gov.justice.laa.crime.orchestration.dto.maat.*;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.FinAssIncomeEvidenceDTO;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.RepOrderDTO;
+import uk.gov.justice.laa.crime.orchestration.dto.maat_api.FinancialAssessmentDTO;
+import uk.gov.justice.laa.crime.util.DateUtil;
+import uk.gov.justice.laa.crime.util.NumberUtils;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.*;
+import java.util.stream.Stream;
+
+import static java.util.Optional.ofNullable;
+
+@Component
+@RequiredArgsConstructor
+public class IncomeEvidenceMapper {
+
+    private final UserMapper userMapper;
+    private final MeansAssessmentMapper meansAssessmentMapper;
+
+    public ApiCreateIncomeEvidenceRequest workflowRequestToApiCreateIncomeEvidenceRequest(WorkflowRequest workflowRequest) {
+        ApplicationDTO application = workflowRequest.getApplicationDTO();
+        ApiApplicantDetails partnerDetails = getPartnerDetails(application.getApplicantLinks());
+        Collection<AssessmentSectionSummaryDTO> initSectionSummaries = application.getAssessmentDTO()
+                .getFinancialAssessmentDTO().getInitial().getSectionSummaries();
+
+        return new ApiCreateIncomeEvidenceRequest()
+                .withMagCourtOutcome(MagCourtOutcome.getFrom(application.getMagsOutcomeDTO().getOutcome()))
+                .withApplicantDetails(getApplicantDetails(application.getApplicantDTO()))
+                .withPartnerDetails(partnerDetails)
+                .withApplicantPensionAmount(getPensionAmount(initSectionSummaries, false))
+                .withPartnerPensionAmount(partnerDetails == null ? null : getPensionAmount(initSectionSummaries, true))
+                .withMetadata(getMetadata(workflowRequest));
+    }
+
+    public MaatApiUpdateAssessment mapToMaatApiUpdateAssessment(WorkflowRequest workflowRequest,
+                                                                RepOrderDTO repOrder,
+                                                                ApiCreateIncomeEvidenceResponse evidenceResponse) {
+        ApplicationDTO application = workflowRequest.getApplicationDTO();
+        String assessmentType = Boolean.TRUE.equals(application.getAssessmentDTO().getFinancialAssessmentDTO().getFullAvailable()) ? "FULL" : "INIT";
```

Comment:
Yeh I almost did this but thought that as we have to use the fully qualified name to specify the financialAssessmentDTO type it made the code more cluttered and less readable. Happy to implement though if you think it would be better like this.